extensions/wikia/PhalanxII/classes/Phalanx.class.php

```diff
@@ -186,44 +280,120 @@ public function delete() {
+	public static function getTypeNames( $typeField ) {
+		$types = [];
+
+		// Start with the largest type bit mask we have, e.g. 0b10000000 = 0x80 = 256
+		$bitMask = self::MAX_TYPE_VALUE;
+
+		while ( $bitMask ) {
+			// If the current bit given by $bitMask is set in $typeField AND we have a
+			// type name for it, save it in $types.  Otherwise skip.
+			if ( $typeField & $bitMask && !empty( self::$typeNames[$bitMask] ) ) {
+				$types[$bitMask] = self::$typeNames[$bitMask];
+			}
+			$bitMask >>= 1;
+		}
 
-		/* iterate for each module for which block is saved */
-		for ( $bit = $typemask & 1, $type = 1; $typemask; $typemask >>= 1, $bit = $typemask & 1, $type <<= 1 ) {
-			if ( !$bit ) continue;
-			$types[$type] = self::$typeNames[$type];
+		// Some data we get will have bits set that are no longer recognized.  If this is the
+		// case at lease output something
+		if ( count( $types ) == 0 ) {
+			$types[0] = 'unknown';
 		}
 
 		return $types;
 	}
 
-	/* get all phalanx types */
+	/**
+	 * Get all phalanx types
+	 *
+	 * @return array
+	 */
 	public static function getAllTypeNames() {
 		return self::$typeNames;
 	}
 
-	/* map array keys to fields in phalanx table */
+	/**
+	 * Get supported phalanx types
+	 *
+	 * @return array
+	 */
+	public static function getSupportedTypeNames() {
+		return array_diff( self::$typeNames, self::$unsupportedTypes );
+	}
+
+	/**
+	 * Mmap array keys to fields in phalanx table
+	 *
+	 * @return array
+	 */
 	private function mapToDB() {
-		$fields = array();
+		$fields = [];
 		foreach ( $this->data as $key => $field  ) {
 			$fields[ 'p_' . $key ] = $field;
 		}
 		return $fields;
 	}
 
+	/**
+	 * If this is an user filter, check if it is not a trusted network IP address
+	 * If it's an IP address, format it correctly
+	 * @return bool true if this user filter is valid, false if it would block trusted host
+	 */
+	private function validateUserFilter(): bool {
+		if ( ( $this->data['type'] & self::TYPE_USER ) && User::isIP( $this->data['text'] ) ) {
+			if ( wfIsTrustedProxy( $this->data['text'] ) ) {
+				// Don't allow to set blocks for trusted proxies or Wikia network hosts
+				return false;
+			}
+
+			$this->data['ip_hex'] = IP::toHex( $this->data['text'] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get connection to Phalanx table on external shared DB
+	 *
+	 * Phalanx table is encoded in utf-8, while in most cases MW communicate with
+	 * databases using latin1, so sometimes we get strings in wrong encoding.
+	 * The only way to force utf-8 communication (adding SET NAMES utf8) is setting
+	 * global variable wgDBmysql5.
+	 *
+	 * @see https://github.com/Wikia/app/blob/dev/includes/db/DatabaseMysqlBase.php#L113
+	 *
+	 * @param int $dbType master or slave
+	 * @return DatabaseBase
+	 */
+	protected function getDatabase( int $dbType = DB_SLAVE ): DatabaseBase {
+		$wrapper = new Wikia\Util\GlobalStateWrapper( [
+			'wgDBmysql5' => true
+		] );
+
+		$db = $wrapper->wrap( function () use ( $dbType ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$db". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)